### DISCUSSION TITLE
Custom plugin has no listeners (iOS) - How to set up the bridge like in android?

### DISCUSSION THREAD

**Discussion by mwilogis** [ionic-team/capacitor#5512] [Q&A] [answered]
I have already created a minimal plugin that is listening for notifications coming from android which is working great. Now I have come to the implementation of iOS. I receive the notification correctly, but when I want to notifyListeners I'm not getting the event. I tried using the hasListeners method, it states I don't have any listeners even though I have. Any help appreciated.

I have been looking at this, to get inspiration:
https://github.com/ionic-team/capacitor-plugins/blob/22fb04179bd13fabea6187c42ed3ffc2faa866c1/push-notifications/ios/Plugin/PushNotificationsHandler.swift

In android I had to set up a bridge to make it work in the load method, but trying to do what the push plugin does in the following link on load does only makes more errors and I don't see that I even how they use the bridge because it only sets up some kind of handler for another class, and not the current one:
https://github.com/ionic-team/capacitor-plugins/blob/22fb04179bd13fabea6187c42ed3ffc2faa866c1/push-notifications/ios/Plugin/PushNotificationsPlugin.swift

I'm trying to figure out swift for the first time, so might be some issues with me not reading the code well enough to see what's really going on.

In the documentation, it not obvious for me how to set the bridge up correctly so I have the right instance:
https://capacitorjs.com/docs/plugins/ios

Setting up listener in ionic:
```
MyPlugin.addListener("ON_EVENT", async ({ event }) => {
        alert("event received!");
      }).catch((error) => {
        alert("event error!");
      });
```

Typescript setting up the plugin:
```
import { registerPlugin, Plugin } from "@capacitor/core";

interface MyCapacitorPluginInterface extends Plugin {}

export const MyPlugin = registerPlugin< MyCapacitorPluginInterface >(
  "MyPlugin"
);
```

In iOS AppDelegate (Receiving the notification):
```
import UIKit
import Capacitor

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

  var window: UIWindow?
  public weak var plugin: CAPPlugin?

  func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
      print("Received notification")
      // let myPlugin: MyPlugin = MyPlugin()
      // myPlugin.OnEvent(eventMessage: "Received notification") <-- also tried calling the plugin class to send it
      let hasListeners = self.plugin?.hasListeners("ON_EVENT")
      print(hasListeners ?? "Has no listeners")
      self.plugin?.notifyListeners("ON_EVENT", data: ["message": "Notification received"])
  }

  ...

```

**Comment by mwilogis** [comment, marked as answer]
Nevermind, just had to use the Notification Center

**Comment by pixelmaxxing** [comment]
How did you exactly define addListener in your customPlugin? I have a plugin which works for calling, but for the love of me I can't get data back from the custom plugin to ionic, addListener is not recognized either
